spec/system/follow_user_pages_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+RSpec.describe "Follow user pages", type: :system, js: true do
+  fab!(:user1) { Fabricate(:user) }
+  fab!(:user2) { Fabricate(:user) }
+  fab!(:user2_post) { Fabricate(:post, user: user2) }
+  let(:everyone_group) { Group[:everyone] }
+
+  before do
+    SiteSetting.discourse_follow_enabled = true
+    SiteSetting.follow_followers_visible = FollowPagesVisibility::EVERYONE
+    SiteSetting.follow_following_visible = FollowPagesVisibility::EVERYONE
+
+    Follow::Updater.new(user1, user2).watch_follow
+    Follow::Updater.new(user2, user1).watch_follow
+  end
+
+  describe 'when user has redesigned user page navigation enabled' do
+    before do
+      everyone_group.add(user1)
+      SiteSetting.enable_new_user_profile_nav_groups = everyone_group.name
+      sign_in(user1)
+    end
+
+    it 'should allow user to navigate to the follow user profile pages' do
+      visit("/u/#{user1.username}")
+
+      click_on "Follows"
```

Comment:
This is only minor and I won't hold up the PR because of it, but I am a little uneasy with adding raw strings to system specs without using `I18n.t`, and I've said that to Nat as well. It creates more places that we have to update. In this situation I would rather use a CSS class or ID.

Here it probably doesn't matter as much since it is for a small plugin (neither does the fact that PageObjects were not used here), but we should keep this in mind and other things when writing system specs, consulting the internal topic about it

Reply:
> but I am a little uneasy with adding raw strings to system specs without using I18n.t

Normally I would use `I18n.t` but the translation does not exist on the server side so you can't use it. Using a CSS class or ID is just as brittle as relying on a client side string. If the client decides to change the CSS class or ID, this test would break. Since that is the case, I prefer to opt for using the raw string which greatly improves readability of the test. 

> (neither does the fact that PageObjects were not used here), but we should keep this in mind and other things when writing system specs, consulting the internal topic about it

I did consult the internal topic but I don't see PageObjects as being useful as this point for the spec because there is only a single test at this point. I do have mostly negative opinions right now about PageObjects but I'll reserve them for the internal topic.

Reply:
> Normally I would use I18n.t but the translation does not exist on the server side so we can't use it. 

In this case we can use `I18n.t("js.blahblah")`, the `js` prefix allows us to access those client translations server-side.

> Using a CSS class or ID is just as brittle as relying on a client side string. If the client decides to change the CSS class or ID, this test would break. Since that is the case, I prefer to opt for using the raw string which greatly improves readability of the test.

>  I do have mostly negative opinions right now about PageObjects but I'll reserve them for the internal topic.

We should definitely discuss these internally to build consensus for consistency, just thought I would point both out. I do agree for this plugin and this small spec a `PageObject` is probably overkill.